src/foam/u2/view/UnstyledTableView.js

```diff
@@ -53,21 +54,28 @@ foam.CLASS({
     },
     {
       name: 'columns_',
+      documentation: `Note: filteredTableColumns was a property set in DAOController
+      for the purpose of having the filtered DAO list available for CSV outputting.`,
       expression: function(columns, of) {
         var of = this.of;
-        if ( ! of ) return [];
-
-        return columns.map(function(p) {
-          var c = typeof p == 'string' ?
-            of.getAxiomByName(p) :
-            p;
-
-           if ( ! c ) {
-             console.error('Unknown table column: ', p);
-           }
-
-          return c;
-        }).filter(function(c) { return c; });
+        this.filteredTableColumns = [];
```

Comment:
Build the array first as a local variable, and then set the imported value afterwards.  This is for two reasons.  1) Anybody listening on updates to filteredTableColumns would first get a property change even as its set to an empty array. and then 2) property change events don't fire when an array's contents change.

Reply:
Actually you'd be better off implementing this as a reaction.

```
foam.CLASS({
  name: 'UnstyledTableView',
...
  imports: [ 'filteredTableColumns?' ],
  reactions: [ '', 'propertyChange.columns_', 'onColumnsUpdate' ],
  listeners: [
    function onColumnsUpdate() {
      // First check if the optional import was provided.
      if ( this.filteredTableColumns$ ) {
        this.filteredTableColumns$ = this.columns_.map(p => p.name);
      }
    }
  ]
});
```

Then you don't need to modify the expression at all, and if somebody explicitly sets columns_ then filteredTableColumns will still be updated appropriately.

Reply:
interesting